csrc/scheduler/utils.cpp

```diff
@@ -1279,8 +1269,7 @@ bool hasInnerDim(
     std::unordered_set<IterDomain*> inner_dims,
     bool should_vectorize) {
   const auto& inner_most_dim = innerMostRootDim(tv);
-  // TODO: Why "|| inner_most_dim->isReduction()"
```

Comment:
`innerMostRootDim` should never return reduction domains? Maybe add an assertion just in case?

Reply:
No, `innerMostRootDim` skips both broadcast and reduction now. `innerMostRootDim` is actually very simple:
```C++
IterDomain* innerMostRootDim(TensorView* tv) {
  const auto& root_domain = tv->getMaybeRFactorDomain();

  if (tv->nDims() == 0) {
    return nullptr;
  }

  IterDomain* inner_most_id = nullptr;

  for (auto it = root_domain.rbegin(); it != root_domain.rend(); it++) {
    if ((*it)->isReduction() || (*it)->isBroadcast()) {
      continue;
    }
    inner_most_id = *it;
    break;
  }

  return inner_most_id;
}
```
I prefer to not assert this, because:
1. This function is simple enough
2. We are not asserting that this function does not return a broadcast, so we should be consistent for reduction.